spanconfig: remove ability to disable span configs

andrewbaptist opened the pull request:
Span configs are integral to correct operation of queues and the ability to disable this infrastructure is no longer necessary.

Epic: none

Release note: None

Commit:
spanconfig: remove ability to disable span configs

Span configs are integral to correct operation of queues and the ability
to disable this infrastructure is no longer necessary.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108168)
<!-- Reviewable:end -->

Commit:
kvserver: pass Desc and SpanConf through allocator

Previously the different layers of the allocator would load the Desc and
SpanConf as needed, this had a risk of them changing between various
loads and could cause strange and hard to track down races. Now they are
loaded once and passed through all the layers.

Epic: none

Release note: None

Commit:
Changes to queues

Commit:
kvserver: removal of cached span config

Epic: none
Fixes: #108123

Release note: None<pkg>: <short description - lowercase, no final period>
